pallet/src/migrations/v0_to_v1.rs

```diff
@@ -0,0 +1,56 @@
+use crate::Config;
+use frame_support::log;
+use frame_support::{
+    traits::{Get, GetStorageVersion, PalletInfoAccess, StorageVersion},
+    weights::Weight,
+};
+use sp_std::prelude::*;
+
+frame_support::generate_storage_alias!(
+    Aleph, OldSessionForValidatorsChange => Value<Option<u32>>
+);
+
+frame_support::generate_storage_alias!(
+    Aleph, OldValidators<T: Config> => Value<Option<Vec<T::AccountId>>>
+);
+
+pub fn migrate<T: Config, P: GetStorageVersion + PalletInfoAccess>() -> Weight {
+    let on_chain_storage_version = <P as GetStorageVersion>::on_chain_storage_version();
+    let new_storage_version = crate::STORAGE_VERSION;
+
+    if on_chain_storage_version == StorageVersion::default() && new_storage_version == 1 {
+        log::info!(target: "pallet_aleph", "Running migration from STORAGE_VERSION 0 to 1");
+
+        let mut writes = 0;
+        let _ = OldSessionForValidatorsChange::translate(|old| match old {
+            Some(current) => {
+                writes += 1;
+                current
+            }
+            None => None,
+        });
+
+        let _ = OldValidators::<T>::translate(|old| match old {
```

Comment:
Can you help me figuring out the mapping between alias `OldValidators` and the actual storage entry? I doubt that it is done by types or naive prefix deduction :P

Reply:
You got me thinking here. The macro signature is :
```
generate_storage_alias!(Prefix, StorageName => Value<u32>);
```

so it must be storing it under:

hex(128-bit hash of `Aleph` + 128-bit hash of `OldValidators`)

the `put_storage_value` does exactly that:
```
pub fn put_storage_value<T: Encode>(module: &[u8], item: &[u8], hash: &[u8], value: T) {
	let mut key = vec![0u8; 32 + hash.len()];
	key[0..16].copy_from_slice(&Twox128::hash(module));
	key[16..32].copy_from_slice(&Twox128::hash(item));
	key[32..].copy_from_slice(hash);
	frame_support::storage::unhashed::put(&key, &value);
}
```
but the the tests were passing even with it. I'll investigate

Reply:
Ok, as per my comment - query(Option(value)) is the same as option-query(value). Translate is therefore an ... identity